src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,246 @@
+            assertThat(member.name).isEqualTo("authentication")
+            assertThat(member.type.isNullable == true)
+        }
+    }
+
+    @Test
+    fun `ensure that no authentication defined will add no parameter`() {
+        val controller = setupTest("global_authentication_none.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).noneSatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).noneSatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+        }
+    }
+
+    @Test
+    fun `ensure that global security and operational security defined will set the correct parameters `() {
+        val controller = setupTest("operation_authentication_with_global.yml")
+
+        val prohibitedController = (controller.find({it.name == "ProhibitedController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val requiredController = (controller.find({it.name == "RequiredController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val optionalController = (controller.find({it.name == "OptionalController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val defaultController = (controller.find({it.name == "DefaultController"})!!.members.single() as TypeSpec).funSpecs.single()
+        val noneController = (controller.find({it.name == "NoneController"})!!.members.single() as TypeSpec).funSpecs.single()
+
+        assertThat(prohibitedController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_ANONYMOUS"}}
+        assertThat(requiredController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}}
+        assertThat(optionalController.annotations).anySatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}}
+        // expect that explicitly set to no authentication
+        assertThat(noneController.annotations).noneSatisfy{member ->
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_ANONYMOUS"}
+            assertThat(member.members).anyMatch{it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}}
```

Comment:
**issue:** These assertions are very smoke-y. I suggest leave the entire "member" logic out here and just assert that there is no `@Secured` annotation at all.

Reply:
done